File: Source/frontend/frontend.Service/ISensorService.cs
using frontend.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace frontend.Service
{
    public interface ISensorService
    {
        List<Sensor> All();
        Sensor Find(int id);
        void Add(Sensor sensor);
        void Update(Sensor sensor);
        void ChangeStatus(Sensor sensor);
    }
}

File: source/back-end/src/main/java/be/pxl/backend/service/ISensorService.java
package be.pxl.backend.service;

import java.util.List;

import be.pxl.backend.entity.Sensor;

public interface ISensorService {
	public Sensor find(int id);
	
	public List<Sensor> all();
	
	public void persist(Sensor sensor);
	
	public void delete(int id);
	
	public void update(Sensor sensor);
	
	public void hardDelete(int id);
}
